geode-core/src/distributedTest/java/org/apache/geode/internal/cache/DeltaPropagationDUnitTest.java

```diff
@@ -1259,93 +1246,107 @@ private void createClientCache(int[] ports, String rLevel, String conflate, Inte
     if (evictAttrs != null) {
       factory.setEvictionAttributes(evictAttrs);
     }
-    if (expAttrs != null) {
-      factory.setEntryTimeToLive(expAttrs);
-    }
+
     if (evictAttrs != null && evictAttrs.getAction().isOverflowToDisk()) {
       // create diskstore with overflow dir
       // since it's overflow, no need to recover, so we can use random number as dir name
       int port = AvailablePort.getRandomAvailablePort(AvailablePort.SOCKET);
       File dir = new File("overflow_" + port);
       if (!dir.exists()) {
-        dir.mkdir();
+        if (!dir.mkdir()) {
+          throw new Exception("mkdir failed");
+        }
       }
       File[] dir1 = new File[] {dir};
       DiskStoreFactory dsf = cache.createDiskStoreFactory();
       factory.setDiskStoreName(dsf.setDiskDirs(dir1).create("client_overflow_ds").getName());
     }
     factory.setConcurrencyChecksEnabled(false);
-    RegionAttributes attrs = factory.create();
+    RegionAttributes<String, DeltaTestImpl> attrs = factory.create();
     cache.createRegion(regionName, attrs);
-    logger = cache.getLogger();
+
   }
 
-  private void createCache(Properties props) throws Exception {
+  private void createCache(Properties props) {
     DistributedSystem ds = getSystem(props);
     ds.disconnect();
     ds = getSystem(props);
-    assertNotNull(ds);
+    assertThat(ds).isNotNull();
     cache = CacheFactory.create(ds);
-    assertNotNull(cache);
+    assertThat(cache).isNotNull();
   }
 
-  private void createDurableCacheClient(Pool poolAttr, String regionName, Properties dsProperties,
-      Integer listenerCode, Boolean close) throws Exception {
+  private void createDurableCacheClient(Pool poolAttr, Properties dsProperties,
+      Boolean close) throws Exception {
     new DeltaPropagationDUnitTest().createCache(dsProperties);
+
     PoolFactoryImpl poolFactory = (PoolFactoryImpl) PoolManager.createFactory();
     poolFactory.init(poolAttr);
-    PoolImpl pool = (PoolImpl) poolFactory.create("DeltaPropagationDUnitTest");
-    AttributesFactory factory = new AttributesFactory();
+    PoolImpl pool = null;
+
+    GeodeAwaitility.await()
```

Comment:
Part two and three of the changes were to move to Awaitility and add better synchronization before doing things...